File: atcoder/abc353/cpp/c.cc
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <tuple>
#include <cassert>

#define rep(i,n) for (long i = 0; i < static_cast<long>((n)); i++)

using namespace std;

int main() {
    long n;
    cin >> n;

    vector<long> a(n);
    rep(i,n) cin >> a[i];

    const long mod = 100000000;
    rep(i,n) a[i] = a[i] % mod;
    sort(a.begin(), a.end());

    long s = 0;
    rep(i,n) s += a[i];
    /* cerr << "s: " << s << endl; */

    long cnt = 0;
    /* rep(i,n) { */
    /*     cerr << "i,a: " << i << " " << a[i] << endl; */
    /* } */
    rep(i,n) {
        auto it = std::lower_bound(a.begin()+i+1, a.end(), mod - a[i]);
        long c = a.end() - it;
        /* cerr << "i,c: " << i << " " << c << endl; */
        cnt += c;
    }

    long ans = (n - 1)*s - cnt*mod;
    cout << ans << endl;
}

File: atcoder/abc353/src/bin/c.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        mut a: [i64; n],
    }
    a.sort();
    let s: i64 = a.iter().sum();

    let m = 100000000;
    let mut l = 0;
    let mut r = n;
    let mut cnt = 0;
    while l < n {
        r = r.max(l + 1);
        while r - l > 1 && a[l] + a[r - 1] >= m {
            r -= 1;
        }
        cnt += n - r;
        l += 1;
    }
    
    let ans = (n - 1) as i64 * s - cnt as i64 *m;
    println!("{ans}");
}
